### PR TITLE
Fix TeamCity project sweeper to run project sweeper only

### PR DESCRIPTION
**Pull request by SarahFrench**
Currently the project sweeper runs all sweepers including the project sweeper.
This PR aims to refactor code so that code is reused to make service sweepers (runs all sweepers bar project sweeper) and project sweepers (run only project sweepers).

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by SarahFrench** [comment]
Hi @roaks3 - I requested you as reviewer because you've got context on project sweepers!

Currently the project sweeper builds in TeamCity run have `SKIP_PROJECT_SWEEPER` unset, but they run all sweepers. This PR makes them more specific.

File: mmv1/third_party/terraform/.teamcity/components/builds/build_configuration_sweepers.kt [hunk 1]
```diff
@@ -10,15 +10,35 @@ import jetbrains.buildServer.configs.kotlin.vcs.GitVcsRoot
...
+    return bc
+}
+
+fun BuildConfigurationForProjectSweeper(providerName: String, sweeperName: String, packages: Map<String, Map<String, String>>, parentProjectName: String, vcsRoot: GitVcsRoot, sharedResources: List<String>, environmentVariables: AccTestConfiguration): BuildType {
+    val sweeperPackage: Map<String, String> = packages.getValue("sweeper")
+    val sweeperPath: String = sweeperPackage.getValue("path").toString()
+
+    val sweeperRun = "GoogleProject" // Name from .google/services/resourcemanager/resource_google_project_sweeper.go
+    val sweeperRegions = "" // Projects aren't region-specific
+
```
[the remark this answers is not in the document]
Comment (roaks3) [reply mmv1/third_party/terraform/.teamcity/components/builds/build_configuration_sweepers.kt:33]: 👍 , if something breaks we could just include the region and it won't be used, but I like the explicitness with keeping it blank.

File: mmv1/third_party/terraform/.teamcity/components/builds/build_parameters.kt [hunk 2]
```diff
@@ -211,7 +218,7 @@ fun BuildType.enableProjectSweep(){
     }
 }
 
-// ParametrizedWithType.terraformEnableProjectSweeper unsets an environment variable used to skip the sweeper for project resources
+// ParametrizedWithType.vcrEnvironmentVariables unsets an environment variable used to skip the sweeper for project resources
```
Comment (roaks3) [inline mmv1/third_party/terraform/.teamcity/components/builds/build_parameters.kt:221]: It seems like maybe the function name in this comment was corrected, but not the rest of it?

**Review by roaks3 — APPROVED** [review]
LGTM, just one nit regarding a comment. I really like the approach of having separate project sweeper runs, and then everything else is just a service sweeper.

File: mmv1/third_party/terraform/.teamcity/components/builds/build_configuration_sweepers.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (SarahFrench) [reply mmv1/third_party/terraform/.teamcity/components/builds/build_configuration_sweepers.kt:33]: I agree, lets push this as it is, and if there's an issue we can iterate on it!

From looking at the project sweeper code I think the region value is [used to set the default provider region in a `transport.Config` struct](https://github.com/hashicorp/terraform-provider-google/blob/a7ba5dc2d0722088daf325abafb9c5b5974a7b6a/google/services/resourcemanager/resource_google_project_sweeper.go#L33C1-L38C3) that influences the API requests made by the sweepers, but the default region doesn't appear to be used. So I'm reasonably confident we can leave it blank, but if I'm wrong there won't be any disruption to the nightly tests.

File: mmv1/third_party/terraform/.teamcity/components/builds/build_parameters.kt (hunk 2 above)
Comment (SarahFrench) [reply mmv1/third_party/terraform/.teamcity/components/builds/build_parameters.kt]: Thanks! I pushed a commit to address this

**Comment by SarahFrench** [comment]
The fix in [543ff68](https://github.com/GoogleCloudPlatform/magic-modules/pull/9940/commits/543ff6881ba42e9eee926887a45be13b27516008) appears to be due to me missing something when copying work from my local TPG repo to my clone of magic-modules.

I'm working on a PR to enable us to run these tests in a GHA, which will catch these problems in future!

**Comment by SarahFrench** [comment]
The expected changes are visible in the UI:

Screenshot 2024-02-08 at 19 18 17

Waiting to see the outcome of this project sweeping build I triggered : https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_PROJECT_SWEEPER_PROJECT_SWEEPER/98460?buildTab=log&logFilter=debug&focusLine=37&logView=linear

**Comment by SarahFrench** [comment]
That ended with:

```
[19:26:24 ] testing: warning: no tests to run
[19:26:24 ]  PASS
[19:26:24 ]  ok    github.com/hashicorp/terraform-provider-google/google/sweeper  0.024s [no tests to run]
[19:26:24 ] Process exited with code 0
```

I've manually triggered a build where sweepRegions = "us-central1", waiting to see the outcome: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_PROJECT_SWEEPER_PROJECT_SWEEPER/98604?branch=%3Cdefault%3E&buildTypeTab=overview&focusLine=NaN&mode=builds&buildTab=overview&hideTestsFromDependencies=false&hideProblemsFromDependencies=false&expandBuildChangesSection=true

If that passes and shows evidence of sweeping in the logs I'll update the code.

**Comment by SarahFrench** [comment]
👆 That build did sweep projects, so providing a value is needed for the sweeper to behave as expected. I've opened a PR for this here: https://github.com/GoogleCloudPlatform/magic-modules/pull/9952

**Pull request merged**